allennlp/testing/test_case.py

```diff
@@ -33,63 +40,47 @@ def setUp(self):
         params['num_epochs'] = 1
-        params['validation_split'] = 0.0
 
         if additional_arguments:
             for key, value in additional_arguments.items():
-                params[key] = deepcopy(value)
+                params[key] = value
         return params
 
-    def get_model(self, model_class, additional_arguments=None):
-        params = self.get_model_params(additional_arguments)
-        return model_class(params)
-
-    def ensure_model_trains_and_loads(self, model_class, args: Params):
-        args['save_models'] = True
-        # Our loading tests work better if you're not using data generators.  Unless you
-        # specifically request it in your test, we'll avoid using them here, and if you _do_ use
-        # them, we'll skip some of the stuff below that isn't compatible.
-        args.setdefault('data_generator', None)
-        model = self.get_model(model_class, args)
-        model.train()
-
-        # load the model that we serialized
-        loaded_model = self.get_model(model_class, args)
-        loaded_model.load_model()
-
-        # verify that original model and the loaded model predict the same outputs
-        if model._uses_data_generators():
-            # We shuffle the data in the data generator.  Instead of making that logic more
-            # complicated, we'll just pass on the loading tests here.  See comment above.
-            pass
-        else:
-            model_predictions = model.model.predict(model.validation_arrays[0])
-            loaded_model_predictions = loaded_model.model.predict(model.validation_arrays[0])
-
-            for model_prediction, loaded_prediction in zip(model_predictions, loaded_model_predictions):
-                assert_allclose(model_prediction, loaded_prediction)
-
-        # We should get the same result if we index the data from the original model and the loaded
-        # model.
-        _, indexed_validation_arrays = loaded_model.load_data_arrays(model.validation_files)
-        if model._uses_data_generators():
-            # As above, we'll just pass on this.
-            pass
-        else:
-            model_predictions = model.model.predict(model.validation_arrays[0])
-            loaded_model_predictions = loaded_model.model.predict(indexed_validation_arrays[0])
-
-            for model_prediction, loaded_prediction in zip(model_predictions, loaded_model_predictions):
-                assert_allclose(model_prediction, loaded_prediction)
+    def ensure_model_saves_and_loads(self,
+                                     model: Model,
+                                     dataset_reader: DatasetReader,
+                                     iterator: DataIterator = None):
+
+        # Our loading tests work better if you're not using complex iterators, so by
+        # default we use the basic one unless you pass an iterator into this function.
+        # If you _do_ use them, we'll skip some of the stuff below that isn't compatible.
```

Comment:
I think the issue here was with shuffling, which is now easy to disable.  But you don't have any checks for them anymore, anyway, so this comment can be removed.